client.go

```diff
@@ -474,27 +477,25 @@ func (c *Client) Do(req *Request) (*http.Response, error) {
 		// Check if we should continue with retries.
 		checkOK, checkErr := c.CheckRetry(req.Context(), resp, err)
 
-		if logger != nil {
-			if err != nil {
+		if err != nil {
+			switch v := logger.(type) {
+			case Logger:
+				v.Printf("[ERR] %s %s request failed: %v", req.Method, req.URL, err)
+			case hclog.Logger:
+				v.Error("request failed", "error", err, "method", req.Method, "url", req.URL)
+			}
+		} else {
+			// Call this here to maintain the behavior of logging all requests,
+			// even if CheckRetry signals to stop.
+			if c.ResponseLogHook != nil {
+				// Call the response logger function if provided.
 				switch v := logger.(type) {
 				case Logger:
-					v.Printf("[ERR] %s %s request failed: %v", req.Method, req.URL, err)
```

Comment:
This line has not been actually modified, it still exists in the same form, see https://github.com/hashicorp/go-retryablehttp/pull/67/files#diff-d0b7f86cea13f90d3cc0cc2338da8d5fR483